File: backend/app/schemas/review.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class ReviewCreate(BaseModel):
    booking_id: uuid.UUID
    overall_rating: int = Field(ge=1, le=5)
    professionalism: int = Field(ge=1, le=5)
    communication: int = Field(ge=1, le=5)
    care_quality: int = Field(ge=1, le=5)
    comment: str | None = Field(default=None, max_length=2000)


class ReviewResponse(BaseModel):
    id: uuid.UUID
    booking_id: uuid.UUID
    patient_id: uuid.UUID
    nurse_id: uuid.UUID
    overall_rating: int
    professionalism: int
    communication: int
    care_quality: int
    comment: str | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: mobile/lib/models/review.dart
class ReviewInfo {
  final String id;
  final String bookingId;
  final String nurseId;
  final int overallRating;
  final int professionalism;
  final int communication;
  final int careQuality;
  final String? comment;
  final DateTime createdAt;

  const ReviewInfo({
    required this.id,
    required this.bookingId,
    required this.nurseId,
    required this.overallRating,
    required this.professionalism,
    required this.communication,
    required this.careQuality,
    this.comment,
    required this.createdAt,
  });

  factory ReviewInfo.fromJson(Map<String, dynamic> j) => ReviewInfo(
        id: j['id'] as String,
        bookingId: j['booking_id'] as String,
        nurseId: j['nurse_id'] as String,
        overallRating: (j['overall_rating'] as num).toInt(),
        professionalism: (j['professionalism'] as num).toInt(),
        communication: (j['communication'] as num).toInt(),
        careQuality: (j['care_quality'] as num).toInt(),
        comment: j['comment'] as String?,
        createdAt: DateTime.parse(j['created_at'] as String),
      );
}
